draft-protected-headers.in.md

```diff
@@ -200,122 +254,120 @@ It is NOT RECOMMENDED to generate messages with such complicated structures.
+The revised algorithm for applying cryptographic protection to a message is as follows:
 
-Typically Visible Headers {#typically-visible-headers}
--------------------------
+- if `crypto` contains encryption, and `legacy` is `true`, and `obscures` contains any user-facing headers (see {{user-facing-headers}}), wrap `orig` in a structure that carries a Legacy Display part:
+  - Create a new MIME leaf part `legacydisplay` with header `Content-Type: text/rfc822-headers; protected-headers="v1"`
+  - For each obscured header name `obh` in `obscures`:
+     - If `obh` is user-facing:
+        - Add `obh: origheaders[ob]` to the body of `legacydisplay`.  For example, if `origheaders['Subject']` is `lunch plans?`, then add the line `Subject: lunch plans?` to the body of `legacydisplay`
+  - Construct a new MIME part `wrapper` with `Content-Type: multipart/mixed`
+  - Give `wrapper` exactly two subarts: `legacydisplay` and `origbody`, in that order.
+  - Let `payload` be MIME part `wrapper`
+- Otherwise:
+  - Let `payload` be MIME part `origbody`
+- For each header name `h` in `origheaders`:
+  - Set header `h` of MIME part `payload` to `origheaders[h]`
+- FIXME: Enigmail adds `protected-headers="v1"` parameter to `payload` here.  Is this necessary?
+- Apply `crypto` to `payload`, producing MIME tree `output`
+- If `crypto` contains encryption:
+  - For each obscured header name `obh` in `obscures`:
+    - If `obscures[obh]` is `null`:
+      - Drop `obh` from `origheaders`
+    - Else:
+      - Set `origheaders[obh]` to `obscures[obh]`
+- For each header name `h` in `origheaders`:
+  - Set header `h` of `output` to `origheaders[h]`
+- return `output`
 
-Of all the headers that an e-mail message may contain, only a handful are typically presented directly to the user.
-The typically visible headers are:
+Note that both new parameters, `obscured` and `legacy`, are effectively ignored if `crypto` does not contain encryption.
+This is by design, because they are irrelevant for signed-only cryptographic protections.
 
- - `Subject`
- - `From`
- - `To`
- - `Cc`
- - `Date`
+Legacy Display {#legacy-display}
+==============
 
-The above is a complete list.  No other headers are considered "typically visible".
+MUAs typically display user-facing headers ({{user-facing-headers}}) directly to the user.
+An encrypted message may be read by a decryption-capable legacy MUA that is unaware of this standard.
+The user of such a legacy client risks losing access to any obscured headers.
 
-Other headers may affect the visible rendering of the message (e.g., `References` and `In-Reply-To` may affect the placement of a message in a threaded discussion), but they are not directly displayed to the user and so are not considered "typically visible" for the purposes of this document.
+This section presents a workaround to mitigate this risk by restructuring the Cryptographic Payload before encrypting to include a "Legacy Display" part.
```

Comment:
s/"Legacy Display"/`Legacy Display`/?

Reply:
I've been trying to use `code` style to indicate literal textual elements, and Upper Case to indicate defined terms.